### PR TITLE
Refine github workflow and freeze some packages

### LINKED ISSUE #363: Run code checks in workflow separately
Currently, it's not easy to see which code check actually failed:
2022-03-09-111549_1441x434_scrot
(here, flake8 failed, not black!)

We should separate code checks into separate steps to make it obvious, for example, whether black or flake8 failed.

### PR DESCRIPTION
**Pull request by jakobj**
As the title suggests.

closes #363

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -20,36 +20,53 @@ jobs:
...
           isort --check-only cgp examples test
-      - name: run tests
+      - name: Run tests
         run: |
+          pytest --version
           pytest --cov=cgp
-      - name: Build Documentation
+      - name: Build documentation
         run: |
-          pip install gym || exit 1;
```
Comment (mschmidt87) [inline .github/workflows/tests.yaml:47]: Is this line not necessary anymore? It seems that you deleted it altogether?

Comment (mschmidt87) [reply .github/workflows/tests.yaml:47]: Ah nevermind, `gym` is already installed via the extra requirements.

**Review by mschmidt87 — COMMENTED** [review]
Is there a particular reason to freeze those packages? Or just a general policy that it's better to freeze instead of being surprise by new releases?

**Comment by jakobj** [comment]
> Is there a particular reason to freeze those packages? Or just a general policy that it's better to freeze instead of being surprise by new releases?

yep, it's necessary to make everything play well together. it's a consequence of having chosen to freeze some packages :( but see #369 for the future of this!

**Review by mschmidt87 — APPROVED** [review]

**Pull request merged**